kubevirt_scripts/poweroff.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
```

Comment:
please add error handling also here and in the other bash scripts

Reply:
I added remaining verifications. I cannot add -eu -o pipefail because:

Some commands may return non-0 status and that's not always a cause to fail. I added verifications when a non-zero must cause the script to exit 1.

A non-zero return code of a piped command doesn't mean it has to fail. I added verifications in such cases as well.

Added set -ux for unset vars and ease debugging.

Reply:
can you please show an example?
maybe you can exclude the error control for those commands adding comments for the reason
same for the pipefail, I'm not sure it's a good idea to remove it, masking errors inside pipes can cause mayhem with apparently no reason
we have multiple long bash scripts here with a lot of "if" blocks that add to complexity, having a good flow and error control would be good

Reply:
For example in mountcd.sh:

~~~sh
NUM_VOLUMES=$(oc -n ${VM_NAMESPACE} get vm ${VM_NAME} -o jsonpath='{.spec.template.spec.volumes[*].name}' | tr " " ";" | grep -o ";" | wc -l)
~~~

If there is only one volume (expected before the first mount happens), tr won't replace any space by a semi-colon. That will cause the piped grep command to not return any values and as such a non-0 return code. If we use pipefail this will exit the execution. Same for -e.

Reply:
Finally added a way to avoid grep causing pipe to fail. Added -o pipefail to the scripts.